packages/auth/src/auth/auth.ts

```diff
@@ -516,10 +521,17 @@ const getCredentialsInternal = async (apiErrorSubStatus?: string) => {
         throw new IllegalArgumentError(authErrorCodeMap.illegalArgumentError);
       }
 
-      if (state.credentials.clientId !== accessToken.clientId) {
+      const shouldUpgradeToken =
+        state.credentials.clientId !== accessToken?.clientId ||
```

Comment:
Not something that should be changed now, but I think the naming could be improved :-) 

I found it strange that `state.credentials` is of type `UserCredentials` and not `Credentials` (which is another type). So I think maybe the thing in `state` could have a different name? (to indicate it includes more than a `Credentials` object would) :-)

Reply:
ye, I agree. That was named before we changed the `Token` to `Credentials`. Naming things is hard 😅